dashboard/src/components/DeploymentForm/DeploymentFormBody/BasicDeploymentForm/TabularSchemaEditorTable/TabularSchemaEditorTable.tsx

```diff
@@ -0,0 +1,251 @@
+// Copyright 2022 the Kubeapps contributors.
+// SPDX-License-Identifier: Apache-2.0
+
+import { CdsButton } from "@cds/react/button";
+import { CdsSelect } from "@cds/react/select";
+import {
+  ColumnFiltersState,
+  ExpandedState,
+  flexRender,
+  getCoreRowModel,
+  getExpandedRowModel,
+  getFacetedMinMaxValues,
+  getFacetedRowModel,
+  getFacetedUniqueValues,
+  getFilteredRowModel,
+  getPaginationRowModel,
+  getSortedRowModel,
+  useReactTable,
+} from "@tanstack/react-table";
+import Column from "components/js/Column";
+import Row from "components/js/Row";
+import LoadingWrapper from "components/LoadingWrapper";
+import { useState } from "react";
+import DebouncedInput from "./DebouncedInput";
+import { fuzzyFilter } from "./TabularSchemaEditorTableHelpers";
+import "./TabularSchemaEditorTable.css";
+import { IBasicFormParam } from "shared/types";
+
+export interface TabularSchemaEditorTableProps {
+  columns: any;
+  data: IBasicFormParam[];
+  globalFilter: any;
+  setGlobalFilter: any;
+  isLoading: boolean;
+  saveAllChanges: () => void;
+}
+
+export default function TabularSchemaEditorTable(props: TabularSchemaEditorTableProps) {
+  const { columns, data, globalFilter, setGlobalFilter, isLoading, saveAllChanges } = props;
+
+  // Component state
+  const [columnFilters, setColumnFilters] = useState<ColumnFiltersState>([]);
+  const [globalExpanded, setGlobalExpanded] = useState<ExpandedState>({});
+
+  const table = useReactTable({
```

Comment:
This is the actual configuration of the Table. More info about the capabilities at https://tanstack.com/table/v8/docs/adapters/react-table